aikau/src/main/resources/alfresco/html/Image.js

```diff
@@ -0,0 +1,421 @@
+            // NOTE: If anyone uses FULL or HASH, don't alter the src path
+            if (this.src) {
+               if (this.srcType === urlTypes.PAGE_RELATIVE) {
+                  this.src = AlfConstants.URL_PAGECONTEXT + this.src.replace(/^\/+/, "");
+               } else if (this.srcType === urlTypes.CONTEXT_RELATIVE) {
+                  this.src = AlfConstants.URL_CONTEXT + this.src.replace(/^\/+/, "");
+               }
+            }
+         },
+
+         /**
+          * This is run after the widget has been created, but before any sub-widgets
+          * have finished being created.
+          *
+          * @instance
+          * @override
+          */
+         postCreate: function alfresco_html_Image__postCreate() {
+            this.inherited(arguments);
+
+            // Add any image classes
+            if (this.classes) {
+               domClass.add(this.imageNode, this.classes);
+            }
+
+            // No configured src, so set to blank.gif if a background image is present
+            if (!this.src) {
+               if (this._getBackgroundImageUrl()) {
+                  this.imageNode.src = require.toUrl(this.pathToBlankGif);
+               } else {
+                  this.imageNode.src = this.src;
+               }
+            }
+
+            // Add "block state"
+            if (this.isBlockElem) {
+               domClass.add(this.domNode, "alfresco-html-Image--block");
+            }
+
+            // Resize the image node
+            this.resize();
+         },
+
+         /**
+          * <p>Get the CSS dimensions of the image node (will not differentiate between dimensions
+          * specified in style attribute and those in a stylesheet). Unavailable and non-numeric
+          * dimensions will be returned as 0.</p>
+          *
+          * <p><strong>NOTE:</strong> If no [src]{@link module:alfresco/html/Image#src} property
+          * has been set, then this will be treated as if no image is present, and so will return
+          * empty (zero) dimensions.</p>
+          *
+          * @instance
+          * @returns {module:alfresco/html/Image#Dimensions} A dimensions object
+          */
+         getCssDimensions: function alfresco_html_Image__getCssDimensions() {
+            var imageStyle = this.imageNode.currentStyle || getComputedStyle(this.imageNode), // IE8 || Proper Browser
+               w = imageStyle.width,
+               h = imageStyle.height;
+            return {
+               w: (w && parseInt(w, 10)) || 0,
+               h: (h && parseInt(h, 10)) || 0
+            };
+         },
+
+         /**
+          * Get the natural size of the currently displayed image.
+          *
+          * @instance
+          * @returns {external:dojo/promise/Promise} A promise that will resolve to a [dimensions]{@link module:alfresco/html/Image#Dimensions} object
+          */
+         getNaturalDimensions: function alfresco_html_Image__getNaturalDimensions() {
```

Comment:
As with earlier comment... if future alignment with Thumbnail is planned, this should be getNaturalImageSize